### PR TITLE
superenv: actually fix calls to -R

### PR DESCRIPTION
**Pull request by jonchang**
Previous attempt: https://github.com/Linuxbrew/brew/pull/328

Possible fix for: https://github.com/Homebrew/homebrew-science/pull/6017

### CODE REVIEW

File: Library/Homebrew/shims/linux/super/cc [hunk 1]
```diff
@@ -189,7 +189,8 @@ class Cmd
     when /^-R(.+)?/
       # -R is not the correct way to pass rpaths to the linker
       val  = chuzzle($1) || enum.next
-      wl = "-Wl," unless mode == :ld
+      path = canonical_path(val)
+      wl   = "-Wl," unless mode == :ld
```
Comment (bw-yuhogg) [inline Library/Homebrew/shims/linux/super/cc]: Can you remove the extra spaces between "wl" and "="?

Comment (jonchang) [reply Library/Homebrew/shims/linux/super/cc]: This matches the style for the rest of the block so I didn't change it.

Comment (bw-yuhogg) [reply Library/Homebrew/shims/linux/super/cc]: Ah, yes it does. Works for me 😄

Comment (sjackman) [reply Library/Homebrew/shims/linux/super/cc]: What's the motivation for using `canonical_path` here? Here's my previous comment on `canonical_path`, which still stands:
https://github.com/Linuxbrew/brew/pull/328#issuecomment-291979672

The error referenced in this issue is
```
shims/linux/super/cc:221:in `keep_orig?': undefined method `start_with?' for nil:NilClass (NoMethodError)
```
https://gist.github.com/ozcan/b5b8f3dd8e9b5dff0c2a066c5a62ef60#file-03-pip-L1505

Comment (sjackman) [reply Library/Homebrew/shims/linux/super/cc]: Yipes. It looks as though this chunk uses `path`, which is never assigned. Does this work?
```diff
diff --git a/Library/Homebrew/shims/linux/super/cc b/Library/Homebrew/shims/linux/super/cc
index 13b6b502b..647914c7c 100755
--- a/Library/Homebrew/shims/linux/super/cc
+++ b/Library/Homebrew/shims/linux/super/cc
@@ -188,7 +188,7 @@ class Cmd
       args << "-L#{val}" if keep?(path) && @lset.add?(path)
     when /^-R(.+)?/
       # -R is not the correct way to pass rpaths to the linker
-      val  = chuzzle($1) || enum.next
+      path  = chuzzle($1) || enum.next
       wl = "-Wl," unless mode == :ld
       args << "#{wl}-rpath=" if keep?(path)
     else
```

**Review by sjackman — CHANGES_REQUESTED** [review]
See comments.

**Comment by jonchang** [comment]
I agree with your suggestion. Comments addressed in a new commit.

File: Library/Homebrew/shims/linux/super/cc [hunk 2]
```diff
@@ -188,8 +188,8 @@ class Cmd
       args << "-L#{val}" if keep?(path) && @lset.add?(path)
     when /^-R(.+)?/
       # -R is not the correct way to pass rpaths to the linker
-      val  = chuzzle($1) || enum.next
-      wl = "-Wl," unless mode == :ld
+      path = chuzzle($1) || enum.next
+      wl   = "-Wl," unless mode == :ld
```
Comment (sjackman) [inline Library/Homebrew/shims/linux/super/cc:192]: I'm not a fan of this style of indentation. I see that nearby code uses it, but I'd rather not propagate it.

**Review by sjackman — CHANGES_REQUESTED** [review]

**Review by sjackman — APPROVED** [review]

**Comment by sjackman** [comment]
Merged. Thanks, Jon!

**Pull request closed without merging**